lib/src/lightclient.rs

```diff
@@ -140,6 +141,7 @@ impl LightClient {
         &self,
         addrs: Vec<(&str, u64, Option<String>)>,
     ) -> Result<String, String> {
+        let transaction_submission_height = self.latest_served_blockheight().await;
```

Comment:
Submission target should be 1 above latest served blockheight, we're aiming for the next block, not the one already mined.